### PR TITLE
adjust to new loading of dynamic key adding to libp2p-crypto

### PR DESCRIPTION
**Pull request by dignifiedquire**
Depends on https://github.com/libp2p/js-libp2p-crypto/pull/74

### CODE REVIEW

**Comment by dignifiedquire** [comment]
@yusefnapora @diasdavid please review this

**Comment by daviddias** [comment]
Sounds like a better approach to DI then the circular dep it had.

Have you confirmed that you can browserify and webpack libp2p-ipfs-browser without having any hurdles?

**Comment by dignifiedquire** [comment]
It works fine with libp2p-crypto, but haven't tested it as a dependency for libp2p-ipfs-browser

File: README.md [hunk 1]
```diff
@@ -40,16 +40,24 @@ npm install --save libp2p-crypto-secp256k1
...
+crypto.keys.secp256k1
+```
+
+Now all methods `generateKeyPair`, `unmarshalPublicKey`, and `marshalPrivateKey` understand `secp256k1`.
 
 ### Example
 
 ```js
 const crypto = require('libp2p-crypto')
+crypto.addKeyType('secp2561k', requrie('libp2p-crypto-secp256k1'))
```
Comment (yusefnapora) [inline README.md:60]: Typo here, "requrie"

**Comment by jacobheun** [comment]
out of date

**Pull request closed without merging**